tensorboard/components/tf_backend/requestManager.ts

```diff
@@ -80,6 +84,19 @@ export class RequestOptions {
    * XMLHttpRequest.withCredentials property.
    */
   public withCredentials?: boolean;
+
+  // Validates this object. Throws InvalidRequestOptionsError on error.
+  public validate() {
+    if (this.methodType === HttpMethodType.GET) {
+      // We don't allow a body for a GET.
+      if (this.body) {
+        throw new InvalidRequestOptionsError(
+          'body must be missing for a GET request.');
+      }
+    }
+    // We allow body-less or contentType-less POSTs even if they don't
```

Comment:
Some validation on POST would be nice. For instance, the body cannot Form when the ContentType is JSON and vice a versa.

Reply:
Hmm. I think it's better if we don't become responsible for enforcing the HTTP protocol standards here, but rather pass the properties opaquely to the XMLHttpRequest.

Reply:
Ok, fair!